src/test/java/emissary/test/core/junit5/ExtractionTest.java

```diff
@@ -171,8 +174,12 @@ protected void checkAnswers(Document answers, IBaseDataObject payload, List<IBas
     }
 
     protected void checkAnswers(Element el, IBaseDataObject payload, List<IBaseDataObject> attachments, String tname) throws DataConversionException {
-
         int numAtt = JDOMUtil.getChildIntValue(el, "numAttachments");
+        long numAttElements = el.getChildren().stream().filter(c -> c.getName().startsWith("att")).count();
+        if (Math.toIntExact(numAttElements) != numAtt && !(numAttElements <= 0 && numAtt == -1) && !skipAttCountValidation) {
+            logger.warn("Expected numAttachments and actual <att#> count in {} not equal. Expected: {} Actual: {}", tname, Math.max(numAtt, 0),
+                    numAttElements);
```

Comment:
Now that we've provided a way to disable validation, we should make this an assertion by default.

```suggestion
            if (numAtt >= -1 || numAttElements > 0) {
                if (!skipAttCountValidation) {
                    assertEquals(numAtt, Math.toIntExact(numAttElements), "Expected numAttachments not equal to number of <att#> elements");
                } else {
                    if (Math.toIntExact(numAttElements) != numAtt) {
                        logger.warn("Expected numAttachments and actual <att#> count in {} not equal. Expected: {} Actual: {}", tname,
                            Math.max(numAtt, 0), numAttElements);
                    }
                }
            }
```

Reply:
so you still want it to log the mismatched counts even with validation skipped? From the previous discussion, it seemed like we wanted the logging to only happen when validation was desired, so that in files where the numbers were super far off, it could be disabled and the warning would not constantly be yelled about.